fiftyone/core/models.py

```diff
@@ -745,7 +745,11 @@ def collate_fn(batch):
                 return error
 
             try:
-                return tud.dataloader.default_collate(batch)
+                return (
+                    tud.dataloader.default_collate(batch)
+                    if not hasattr(model, "collate_fn")
```

Comment:
Looks like this PR needs to be rebased onto https://github.com/voxel51/fiftyone/pull/5834, as this isn't how the final implementation of `collate_fn` landed.